src/ir/effects.h

```diff
@@ -454,7 +433,16 @@ struct EffectAnalyzer
         case DivUInt64:
         case RemSInt64:
         case RemUInt64: {
+          // div and rem may contain implicit trap only if RHS is
+          // non-constant or constant which equal zero or -1 for
+          // signed divisions. Reminder traps only with zero
+          // divider.
           implicitTrap = true;
+          if (auto* c = curr->right->dynCast<Const>()) {
+            implicitTrap = c->value.isZero() ||
```

Comment:
Good idea, but this nesting is not that readable, I think. How about
```
if (c->value.isZero()) {
  implicitTrap = true;
} else if (..) {
```
?

Reply:
Alright, I refactored this